metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -511,3 +501,90 @@ def _convert_value(param):
         "(run-id *{run_id}*).".format(name=obj.workflow_name, run_id=run_id),
         bold=True,
     )
+
+
+@argo_workflows.command(help="Delete the Argo workflow")
+@click.option(
+    "--authorize",
+    default=None,
+    type=str,
+    help="Authorize the deletion with a production token",
+)
+@click.pass_obj
+def delete(obj, authorize=None):
+    validate_token(obj.workflow_name, obj.token_prefix, obj, authorize, "delete")
+    obj.echo("Deleting workflow *{name}*...".format(name=obj.workflow_name))
+
+    # Always try to delete the schedule. Failure in deleting the schedule should not
+    # be treated as an error, due to any of the following reasons
+    # - there might not have been a schedule, or it was deleted by some other means
+    # - retaining these resources should have no consequences as long as the workflow deletion succeeds.
+    # - regarding cost and compute, the significant resources are part of the workflow teardown, not the schedule.
+    schedule_deleted = ArgoWorkflows.remove_schedule(obj.workflow_name)
+    if schedule_deleted:
+        obj.echo("   Removed schedule")
+
+    # After cleaning up related resources, delete the workflow in question.
+    # Failure in deleting is treated as critical and will be made visible to the user
+    # for further action.
+    workflow_deleted = ArgoWorkflows.delete(obj.workflow_name)
+    if workflow_deleted:
+        obj.echo(
+            "   Workflow is being deleted\n"
```

Comment:
A running workflow object would be present in the cluster if a cronworkflow is running when the delete command is issued. That running workflow will neither be interrupted nor deleted with this `delete` command.

Should we add that info to this message? Something like:

```
Deletion of resource might take a while.
Currently running workflows and previously run workflows may have to be deleted manually.
Deploying a flow ...
```

Reply:
added a line to the output describing the effect that delete has on running workflows so the outcome should be more clear now.